mayastor/src/bdev/nexus/nexus_bdev_children.rs

```diff
@@ -422,6 +423,56 @@ impl Nexus {
                     }
                 }
             }
+            DeviceEventType::AdminCommandCompletionFailed => {
+                let cn = &device;
+                for nexus in nexus::instances() {
+                    if nexus
+                        .children
+                        .iter()
+                        .filter(|c| c.state() == ChildState::Open)
```

Comment:
what if the child was rebuilding? So it's state would be Faulted(OutOfSync) IIRC?

Reply:
Good point. This code that marks the child as `Faulted` is basically from `NexusChannelInner::fault_child()` but we can't easily call that from here because we don't have a channel as we're not performing IO. Presumably it'll just ignore the child and as the child is rebuilding anyway the failure will be noticed in the IO path and acted upon there.

Reply:
I've factored out the iterating over the children of a Nexus to mark it as faulted but I've left its functionality alone. A child being rebuilt, which then becomes inaccessible, is a separate case and I'm inclined to only leave this as fixing the this issue whilst ensuring the original issue of handling admin command completion failure remains fixed, without having to fix all possible issues in the area.